pkg/workflows/wasm/host/module.go

```diff
@@ -366,22 +400,27 @@ func (m *Module) Run(request *wasmpb.Request) (*wasmpb.Response, error) {
 	_, err = start.Call(store)
 	switch {
 	case containsCode(err, wasm.CodeSuccess):
-		resp, innerErr := m.r.get(request.Id)
+		storedRequest, innerErr := m.requestStore.pop(request.Id)
```

Comment:
@agparadiso I don't think this is enough to remove the job in all cases: if we fall through any of the other cases the entry we added will still remain.

Probably a better way to handle this is to move the pop so that it's global, i.e. move to just after line 400.

Then handle the error, and where you need to use a response, you can

Reply:
good catch, I switched the implementation to have a `delete` method instead of `pop` and `defer delete()` after adding it to the store